driver/main.c

```diff
@@ -2972,7 +2972,15 @@ static int set_g_buffer_bytes_dim(const char *val, const struct kernel_param *kp
 	return param_set_ulong(val, kp);
 }
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 36)
+static const struct kernel_param_ops g_buffer_bytes_dim_ops = {
+	.set	= set_g_buffer_bytes_dim,
+	.get	= param_get_ulong,
+};
+module_param_cb(g_buffer_bytes_dim, &g_buffer_bytes_dim_ops, &g_buffer_bytes_dim, 0644);
+#else
 module_param_call(g_buffer_bytes_dim, set_g_buffer_bytes_dim, param_get_ulong, &g_buffer_bytes_dim, 0644);
+#endif
```

Comment:
Here we have introduced the `kernel_param_ops` because it has a stable interface https://elixir.bootlin.com/linux/v2.6.36/source/include/linux/moduleparam.h#L34
The problem with using directly the `module_param_call` is that the prototype of the `set` method is changed over time
 in some kernel versions (like 2.6.36) is:

 int (*[oldset](https://elixir.bootlin.com/linux/v2.6.36/C/ident/oldset))(const char *, struct [kernel_param](https://elixir.bootlin.com/linux/v2.6.36/C/ident/kernel_param) *);

like in recent kernel versions like (5.11) we pass through the `kernel_param_ops` as you can see here https://elixir.bootlin.com/linux/v5.11/source/include/linux/moduleparam.h#L297, so we have this signature:

int (*[set](https://elixir.bootlin.com/linux/v5.11/C/ident/set))(const char *val, const struct [kernel_param](https://elixir.bootlin.com/linux/v5.11/C/ident/kernel_param) *[kp](https://elixir.bootlin.com/linux/v5.11/C/ident/kp));

Reply:
So using always something like `static int set_g_buffer_bytes_dim(const char *val, const struct kernel_param *kp)` we have no problems because in kernels <= `2.6.36` the `const` qualifier won't be used but the build will be successful